indexer/rank_table.hpp

```diff
@@ -56,8 +56,11 @@ class RankTable
   // Returns underlying data format version.
   virtual Version GetVersion() const = 0;
 
-  // Serializes rank table.
-  virtual void Serialize(Writer & writer) = 0;
+  // Serializes rank table. When |preserveHostEndianness| is true,
+  // table is serialized in host endianness, otherwise, opposite
+  // endianness is used. Please, don't set this parameter to false
+  // unless you know what you do.
+  virtual void Serialize(Writer & writer, bool preserveHostEndianness) = 0;
```

Comment:
Because we're not interested here in big/little endianness - we're interested in preserving host machine's endianness or not. It can be Little Endian or Big Endian - it doesn't matter. So one should read this method as "serialize this for the same architectures as the host is or serialize this for foreign architectures." Explicit endianness will cause difficulties in using this method as clients should always remember that we're Little Endian only. Moreover, this unit is designed in a such way that it's even possible to deserialize this data structure from foreign endianness, thus, the main reason to introduce the last parameter was for testing purposes.